src/withServiceWorkerUpdater.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, forwardRef, useEffect } from 'react'
+import updateSW from './updateSW'
+
+export interface ServiceWorkerUpdaterProps {
+  newServiceWorkerDetected: boolean
+  onLoadNewServiceWorkerAccept: () => void
+}
+
+export interface WithServiceWorkerUpdaterOptions {
+  message?: unknown
```

Comment:
Not sure if there's a specific type this should be associated with.

Reply:
- `message`: [This can be any structured-clonable type](https://developer.mozilla.org/en-US/docs/Web/API/Client/postMessage#parameters)
- [structured-clonable type](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)

It looks pretty `unknown` (any type of simple or complex data, but not functions or classes). I'm not TypeScript expert and maybe exists something specific for this.